awx/main/tasks/receptor.py

```diff
@@ -0,0 +1,318 @@
+# receptor.py
+# Python
+from base64 import b64encode
+from collections import namedtuple
+import concurrent.futures
+import socket
+import sys
+import threading
+import time
+import yaml
+
+# Django
+from django.conf import settings
+
+
+# Runner
+import ansible_runner
+import ansible_runner.cleanup
+from awx.main.utils.common import (
+    deepmerge,
+    parse_yaml_or_json,
+    cleanup_new_process,
+)
+
+
+# AWX
+from awx.main.utils.receptor import get_receptor_ctl, get_conn_type, get_tls_client
```

Comment:
It's likely possible to move code from `awx.main.utils.receptor` into this file, and that would make the most sense to me.